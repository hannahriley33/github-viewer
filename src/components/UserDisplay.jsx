import React from 'react';
import PropTypes from 'prop-types';

const UserDisplay = ({ name, followers, following, url }) => (
  <li>
    <p>{name}</p>
    <p>{followers}</p>
    <p>{following}</p>
    <a href={url}> Link to Github profile </a>
  </li>
);

UserDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired
};

export default UserDisplay;
