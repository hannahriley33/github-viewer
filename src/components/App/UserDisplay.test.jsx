import React from 'react';
import { shallow } from 'enzyme';
import UserDisplay from '../UserDisplay';

describe('UserDisplay', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UserDisplay name="Rick" followers="2" following="3" url="https://rickandmortyapi.com/api/character/avatar/1.jpeg" />);
    expect(wrapper).toMatchSnapshot();
  });
});
