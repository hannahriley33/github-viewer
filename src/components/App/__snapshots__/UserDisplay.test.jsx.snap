// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay matches a snapshot 1`] = `
<li>
  <p>
    Rick
  </p>
  <p>
    2
  </p>
  <p>
    3
  </p>
  <a
    href="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  >
     Link to Github profile 
  </a>
</li>
`;
